scripts/magpietts/update_magpie_shars.py

```diff
@@ -0,0 +1,245 @@
+                    fields=self.fields,
+                    shard_size=self.shard_size,
+                )
+                self.shar_writer.__enter__()
+
+            self.is_initialized = True
+
+    def process_cut_for_saving(self, cut: MonoCut) -> MonoCut:
+        new_custom = {}
+        for k, v in cut.custom.items():
+            if k in self.fields:
+                new_custom[k] = v
+        
+        cut.custom = new_custom
+        return cut
+
+    def write_on_batch_end(self, trainer, pl_module, prediction, batch_indices, batch, batch_idx, dataloader_idx):
+        import time
+        st = time.time()
+        cuts_local, codes_local, context_codes_local = prediction["cuts"], prediction["codes"], prediction["context_codes"]
+        world_size = trainer.world_size
+        if world_size > 1:
+            gathered = [None] * trainer.world_size
+            torch.distributed.all_gather_object(gathered, (cuts_local, codes_local, context_codes_local))
+        else:
+            gathered = [(cuts_local, codes_local, context_codes_local)]
+
+        if trainer.global_rank == 0:
+            for cuts_rank, codes_rank, context_codes_rank in gathered:
+                for cut, code, context_code in zip(cuts_rank, codes_rank, context_codes_rank):
+                    cut = cut.attach_tensor(
+                        name=f"codes_{self.codec_model_name}",
+                        data=code.cpu().type(torch.int16).numpy(),
+                    ).attach_tensor(
+                        name=f"context_codes_{self.codec_model_name}",
+                        data=context_code.cpu().type(torch.int16).numpy(),
+                    )
+                    self.cuts_buffer.append(cut)
+
+            if len(self.cuts_buffer) >= self.buffer_size:
+                self._write_buffer()
+            
+        et = time.time()
+        print("Time to write:", et - st)
+
+    def _write_buffer(self):
+        print("Writing buffer of size:", len(self.cuts_buffer))
+        for cut in self.cuts_buffer:
+            # print("Writing cut:", cut.id)
+            processed_cut = self.process_cut_for_saving(cut)
+            self.shar_writer.write(processed_cut)
+        self.cuts_buffer.clear()
+
+    def teardown(self, trainer, pl_module, stage: str | None = None):
+        if trainer.world_size > 1:
+            torch.distributed.barrier()
+
+        if trainer.global_rank == 0 and self.shar_writer is not None:
+            if self.cuts_buffer:
+                self._write_buffer()
+            self.shar_writer.close()
+            self.shar_writer = None
+
+def ddp_info():
+    if torch.distributed.is_available() and torch.distributed.is_initialized():
+        return torch.distributed.get_world_size(), torch.distributed.get_rank()
+    return 1, 0
+
+class CodecExtractor(pl.LightningModule):
+    def __init__(self, model_path, pad_multiple, batch_size, shar_root, num_workers):
+        super().__init__()
+        self.pad_multiple = pad_multiple
```

Comment:
we don't need this argument anymore since we can always get it from `self.codec_model.samples_per_frame`

Reply:
That's right, changed this.